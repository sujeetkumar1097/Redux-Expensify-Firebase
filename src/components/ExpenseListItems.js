import React from 'react';
import {NavLink} from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'
 
const ExpenseListItems = ({ id, description, amount, createdAt}) => {
    //console.log("this is expense: obj ==>>",id);
    return (
            <NavLink className="list-item" to={`/edit/${id}`}>
                <div>
                    <h3 className="list-item__title">{description}</h3>
                    <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
                </div>
                <h3 className="list-item__amount">{numeral(amount/100).format('$0,0.00')}</h3>
            </NavLink>
    )
}
export default ExpenseListItems

