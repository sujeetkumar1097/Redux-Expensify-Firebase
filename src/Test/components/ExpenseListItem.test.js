import React from 'react'
import {shallow} from 'enzyme'
import  ExpenseListItems from '../../components/ExpenseListItems'
import expenses from '../Fixtures/Expenses'

test('Expense List Render Correctly', () =>{
    const wrapper = shallow(<ExpenseListItems expenses={expenses[0]}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Expense List Render Correctly', () =>{
    const wrapper = shallow(<ExpenseListItems expenses={[]}/>)
    expect(wrapper).toMatchSnapshot()
})