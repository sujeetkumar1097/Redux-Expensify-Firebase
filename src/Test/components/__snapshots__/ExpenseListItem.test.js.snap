// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List Render Correctly 1`] = `
<NavLink
  activeClassName="active"
  aria-current="page"
  className="list-item"
  to="/edit/undefined"
>
  <div>
    <h3
      className="list-item__title"
    />
    <span
      className="list-item__sub-title"
    >
      January 1st, 1970
    </span>
  </div>
  <h3
    className="list-item__amount"
  >
    $0.00
  </h3>
</NavLink>
`;

exports[`Expense List Render Correctly 2`] = `
<NavLink
  activeClassName="active"
  aria-current="page"
  className="list-item"
  to="/edit/undefined"
>
  <div>
    <h3
      className="list-item__title"
    />
    <span
      className="list-item__sub-title"
    >
      January 1st, 1970
    </span>
  </div>
  <h3
    className="list-item__amount"
  >
    $0.00
  </h3>
</NavLink>
`;
